### PR TITLE
feat: support setting a filter in autoscaling metrics

### PR DESCRIPTION
**Pull request by jbuet**
Enable the filter setting in the autoscaling_metric. It's required to use the google-beta provider for this field https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_region_autoscaler#filter

I can edit commit and PR title appropriately.

### CODE REVIEW

**Comment by jbuet** [comment]
@apeabody hello! can you help me by reviewing this PR? thank you!

File: modules/mig/main.tf [hunk 1]
```diff
@@ -117,7 +117,7 @@ resource "google_compute_region_instance_group_manager" "mig" {
 }
 
 resource "google_compute_region_autoscaler" "autoscaler" {
-  provider = google
+  provider = google-beta
```
Comment (apeabody) [inline modules/mig/main.tf:120]: Is this going to be a breaking change for existing clusters?  Please also verify if any updated is needed to the provider minimum version: https://github.com/terraform-google-modules/terraform-google-vm/blob/master/modules/mig/versions.tf#L26

File: modules/mig_with_percent/main.tf [hunk 2]
```diff
@@ -126,7 +126,7 @@ resource "google_compute_region_instance_group_manager" "mig_with_percent" {
 }
 
 resource "google_compute_region_autoscaler" "autoscaler" {
-  provider = google
+  provider = google-beta
```
Comment (apeabody) [inline modules/mig_with_percent/main.tf:129]: same as above

**Review by apeabody — CHANGES_REQUESTED** [review]
Thanks for the contribution @jbuet!  Please see the question/comment below.

**Pull request closed without merging**